src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,253 @@
+            Scanner in = new Scanner(System.in);
+            String line = in.nextLine();
+
+            // process user input and decide what operation to use
+            String operation = extractOperationType(line);
+            String printStatement;
+
+            switch (operation){
+                case "bye":
+                    hasExit = true;
+                    printStatement = printByeMessage();
+                    break;
+                case "list":
+                    printStatement = printFullList(numberOfTasks);
+                    break;
+
+                case "done":
+                    try {
+                        printStatement = markTaskAsDone(line);
+                    } catch ( IllegalIndex e) {
+                        returnIllegalIndexStatement();
+                        continue;
+                    }
+                    break;
+                case "delete":
+                    try {
+                        printStatement = deleteTask(line);
+                    } catch ( IllegalIndex e) {
+                        returnIllegalIndexStatement();
+                        continue;
+                    }
+                    break;
+
+                case "todo":
+                case "deadline":
+                case "event":
+                    try {
+                        if(operation.equals("todo")) printStatement = createNewTask(operation,extractDescriptionFromString(operation,line),null,false);
+                        else printStatement = createNewTask(operation,extractDescriptionFromString(operation,line),extractDateFromString(line),false);
+                    } catch (IllegalDate e) {
+                        printEmptyDate();
+                        continue;
+                    } catch (IllegalDescription e){
+                        printEmptyDescription();
+                        continue;
+                    }
+                    break;
+                default:
+                    printUnknownMessage();
+                    continue;
+
+            }
+            System.out.println(printStatement);
+            save.writeFile(home+"/Documents/log.txt",printFullList(numberOfTasks));
+        }
+    }
+    
+    public static String createNewTask(String taskType, String description, String date, Boolean status) throws IllegalDate, IllegalDescription {
+        String printStatement = null;
+        if(description.isEmpty()) throw new IllegalDescription();
+        switch(taskType) {
+            case "todo":
+
+                ToDo t = new ToDo(description);
+                list.add(t);
+                printStatement = printNIncrementTask(numberOfTasks);
+                break;
+
+            case "deadline":
+                Deadline d = new Deadline(description, date);
+                list.add(d);
+                printStatement = printNIncrementTask(numberOfTasks);
+                break;
+
+            case "event":
+                Event e = new Event(description, date);
```

Comment:
Perhaps can use a more descriptive variable name here.

Reply:
So something like "Event event = new Event(description, date)"?

Reply:
Yup that can something like that. I was just concerned about e being too vague because while the context is different, "e" was used in one of the catch statement (catch (exception e)). So its good to just specify the variable names👍